apps/web/components/search/search-dialog.tsx

```diff
@@ -0,0 +1,421 @@
+"use client";
+
+import React from "react";
```

Comment:
```suggestion
import type { KeyboardEvent } from "react";
```
Explicitly import the type or property into the component, Don't use the global import